### PR TITLE
Upgrade ruby to 3.1.2

### PR DESCRIPTION
**Pull request by callumknights**
Upgrades Ruby to 3.1.2. See commits for more details.
This PR should be tested before merge, as per https://docs.publishing.service.gov.uk/manual/ruby.html#update-ruby-version-in-the-relevant-repos.

Trello: https://trello.com/c/nqtTBLJK/347-upgrade-ruby-for-sidekiq-monitoring

⚠️ This repo is Continuously Deployed: make sure you [follow the guidance](https://docs.publishing.service.gov.uk/manual/development-pipeline.html#merge-your-own-pull-request) ⚠️

### CODE REVIEW

**Comment by callumknights** [comment]
Checked on Integration and the app runs as intended

File: .ruby-version [hunk 1]
```diff
@@ -1 +1,2 @@
-2.7.6
+3.1.2
+
```
Comment (brucebolt) [inline .ruby-version:2]: Looks like there's an extra newline here that should probably be removed

Comment (kevindew) [inline .ruby-version:2]: Looks like we have an extra line here?
```suggestion
```

**Review by kevindew — APPROVED** [review]
Looks good, just a small fix needed

File: .ruby-version (hunk 1 above)
Comment (callumknights) [reply .ruby-version]: Thanks for spotting, sorted 👍

**Pull request merged**